src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+            return tt0
+        if len(barycentricTOA) > 1:
+            if len(self.piecewise_parameter_information) > 0:
+                self.toa_belongs_in_group(
+                    barycentricTOA
+                )  # Defines object's index for each toa as an array of length = len(self._t)
+                (
+                    self.T0X_per_toa,
+                    self.A1X_per_toa,
+                ) = self.piecewise_parameter_from_information_array(
+                    self._t
+                )  # Uses the index for each toa array to create arrays where elements are the A1X/T0X to use with that toa
+        if len(barycentricTOA) > 1:
+            check = hasattr(self, "T0X_per_toa")
+            if hasattr(self, "T0X_per_toa") is True:
+                if len(self.T0X_per_toa) == 1:
+                    T0 = self.T0X_per_toa
+                    # print("hello from 1")
+                else:
+                    T0 = self.T0X_per_toa
+                    # print("hello from 2")
+            else:
+                T0 = self.T0
+                # print("hello from 3")
+        else:
+            T0 = self.T0
+            # print("hello from 4")
+        if not hasattr(barycentricTOA, "unit") or barycentricTOA.unit == None:
+            barycentricTOA = barycentricTOA * u.day
+        # print(f"Unique T0s being used in tt0 calculation: {np.unique(T0)}\n")
+        tt0 = (barycentricTOA - T0).to("second")
+        # print(f"piecewise tt0 being called, Calc'd with: {T0} ")
+        return tt0
+
+    def d_delayL1_d_par(self, par):
+        if par not in self.binary_params:
+            errorMesg = par + " is not in binary parameter list."
+            raise ValueError(errorMesg)
+        par_obj = getattr(self, par)
+        index, par_temp = self.in_piece(par)
+        # print(index)
+        if (
+            par_temp is None
+        ):  # to circumvent the need for list of d_pars = [T0X_0,...,T0X_i] use par_temp
+            if hasattr(self, "d_delayL1_d_" + par):
+                func = getattr(self, "d_delayL1_d_" + par)
+                return func() * index
+            else:
+                if par in self.orbits_cls.orbit_params:
+                    return self.d_delayL1_d_E() * self.d_E_d_par(par)
+                else:
+                    return np.zeros(len(self.t)) * u.second / par_obj.unit
+        else:
+            if hasattr(self, "d_delayL1_d_" + par_temp):
+                func = getattr(self, "d_delayL1_d_" + par_temp)
+                return func() * index
+            else:
+                if par in self.orbits_cls.orbit_params:
+                    return self.d_delayL1_d_E() * self.d_E_d_par()
+                else:
+                    return np.zeros(len(self.t)) * u.second / par_obj.unit
+
+    def d_delayL2_d_par(self, par):
+        if par not in self.binary_params:
+            errorMesg = par + " is not in binary parameter list."
+            raise ValueError(errorMesg)
+        # print(par)
```

Comment:
Please deal with debugging `print` statements - either delete them, or if you think they might be useful for debugging, use `log.debug`.

Reply:
All gone now!